modules/context/repo.go

```diff
@@ -49,23 +49,25 @@ type PullRequest struct {
 // Repository contains information to operate a repository
 type Repository struct {
 	models.Permission
-	IsWatching   bool
-	IsViewBranch bool
-	IsViewTag    bool
-	IsViewCommit bool
-	Repository   *models.Repository
-	Owner        *models.User
-	Commit       *git.Commit
-	Tag          *git.Tag
-	GitRepo      *git.Repository
-	BranchName   string
-	TagName      string
-	TreePath     string
-	CommitID     string
-	RepoLink     string
-	CloneLink    models.CloneLink
-	CommitsCount int64
-	Mirror       *models.Mirror
+	IsWatching        bool
+	IsViewBranch      bool
+	IsViewTag         bool
+	IsViewCommit      bool
+	Repository        *models.Repository
+	Owner             *models.User
+	Commit            *git.Commit
+	Tag               *git.Tag
+	GitRepo           *git.Repository
+	BranchName        string
+	IsRenamedBranch   bool
```

Comment:
I don't think we should put more and more fields in this struct, only those which could be used by all `{username}/{reponame}` routes should be stored on the struct. You can store them into `Data`.